testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/AbstractHttpClientTest.java

```diff
@@ -1151,6 +1153,21 @@ protected SpanDataAssert assertClientSpan(
               } else {
                 assertThat(attrs).doesNotContainKey(HttpAttributes.HTTP_REQUEST_RESEND_COUNT);
               }
+            })
+        .satisfies(
+            spanData -> {
+              // Check for peer.service when running with javaagent instrumentation
+              String distroName =
+                  spanData
+                      .getResource()
+                      .getAttribute(TelemetryIncubatingAttributes.TELEMETRY_DISTRO_NAME);
+              if ("opentelemetry-java-instrumentation".equals(distroName)) {
+                String expectedPeerService = options.getExpectedPeerServiceName().apply(uri);
+                if (expectedPeerService != null) {
+                  assertThat(spanData.getAttributes())
+                      .containsEntry(PeerIncubatingAttributes.PEER_SERVICE, expectedPeerService);
+                }
+              }
```

Comment:
I didn't want to deal with configuring all of the library http client telemetry builders, but now that the javaagent ones are working, I don't mind revisiting

(in general I don't think `peer.service` infra is as important for library instrumentation because it's generally easy to add a `peer.service` constant identifier when you add instrumentation around a particular service call in that case)